portal/v2/src/ClusterDetailListComponents/ClusterOperatorsWrapper.tsx

```diff
@@ -0,0 +1,100 @@
+import { IMessageBarStyles, MessageBar, MessageBarType, Stack } from '@fluentui/react';
+import { clusterOperatorsKey } from "../ClusterDetail";
+import { ClusterOperatorListComponent } from "./ClusterOperatorList";
+
+export interface ICondition {
+  status: string,
+  reason: string,
+  lastUpdated: string,
+  message: string
+}
+
+export interface IClusterOperator {
+  name: string,
+  available: string,
+  conditions?: ICondition[],
+}
+
+export function ClusterOperatorsWrapper(props: {
+  currentCluster: ICluster
+  detailPanelSelected: string
+  loaded: boolean
+}) {
+  const [data, setData] = useState<any>([])
+  const [error, setError] = useState<AxiosResponse | null>(null)
+  const state = useRef<ClusterOperatorListComponent>(null)
+  
+  const [fetching, setFetching] = useState("")
+
+  const errorBarStyles: Partial<IMessageBarStyles> = { root: { marginBottom: 15 } }
+
+  const errorBar = (): any => {
+    return (
+      <MessageBar
+        messageBarType={MessageBarType.error}
+        isMultiline={false}
+        onDismiss={() => setError(null)}
+        dismissButtonAriaLabel="Close"
+        styles={errorBarStyles}
+      >
+        {error?.statusText}
+      </MessageBar>
+    )
+  }
+
+  // updateData - updates the state of the component
+  // can be used if we want a refresh button.
+  // api/clusterdetail returns a single item.
+  const updateData = (newData: any) => {
+    setData(newData)
+    const clusterOperatorList: IClusterOperator[] = []
+    if (state && state.current) {
+      newData.operators.forEach((element: { name: any;
+                                        available: any;
+                                        conditions: ICondition[]}) => {
+        const clusterOp: IClusterOperator = {
+          name: element.name,
+          available: element.available
+        }
+        clusterOp.conditions = []
+        element.conditions.forEach((condition: ICondition) => {
+          clusterOp.conditions!.push(condition)
+        });
+        clusterOperatorList.push(clusterOp)
+      });
+      state.current.setState({ clusterOperators: clusterOperatorList })
+    }
+  }
+
+  useEffect(() => {
+    const onData = (result: AxiosResponse | null) => {
+      if (result?.status === 200) {
+        updateData(result.data)
+      } else {
+        setError(result)
+      }
+      setFetching(props.currentCluster.name)
+    }
+
+    if (props.detailPanelSelected.toLowerCase() == clusterOperatorsKey && 
+        fetching === "" &&
+        props.loaded &&
+        props.currentCluster.name != "") {
+      setFetching("FETCHING")
+      FetchClusterOperators(props.currentCluster).then(onData)
+    }
+  }, [data, props.loaded, props.detailPanelSelected])
+  
+  return (
+    <Stack>
+      <Stack.Item grow>{error && errorBar()}</Stack.Item>
+      <Stack>
+        <ClusterOperatorListComponent clusterOperators={data!} ref={state} clusterName={props.currentCluster != null ? props.currentCluster.name : ""} />
```

Comment:
Can we actually assume that we have `data` here at this point? If we have an `error` will that mean that `data` may be undefined?

Reply:
Was able to just remove this since we use an array now it can be empty